### PR TITLE
NXP Hal update to SDK 2.13

### PR DESCRIPTION
**Pull request by yvanderv**
New device RT1042, code update.

### CODE REVIEW

File: mcux/mcux-sdk/drivers/lpuart/fsl_lpuart.h [hunk 1]
```diff
@@ -650,9 +650,9 @@ uint32_t LPUART_GetEnabledInterrupts(LPUART_Type *base);
  * @param base LPUART peripheral base address.
  * @return LPUART data register addresses which are used both by the transmitter and receiver.
  */
-static inline uint32_t LPUART_GetDataRegisterAddress(LPUART_Type *base)
+static inline uintptr_t LPUART_GetDataRegisterAddress(LPUART_Type *base)
 {
-    return (uint32_t) & (base->DATA);
+    return (uintptr_t) & (base->DATA);
```
Comment (dleach02) [inline mcux/mcux-sdk/drivers/lpuart/fsl_lpuart.h]: Need to document this in the mcux/README in the patch list. Also, submit this back to the SDK team so we don't have to keep changing.

File: mcux/mcux-sdk/LA_OPT_NXP_Software_License.txt
Comment (dleach02) [inline mcux/mcux-sdk/LA_OPT_NXP_Software_License.txt:1]: What is being changed in this file?

**Comment by DerekSnell** [comment]
Hi @matt-wood-ct ,

NXP plans to add board support for the MIMXRT1040-EVK to the Zephyr repo in time for the Zephyr v3.4 release.  When the PR is created for the initial support, you are very welcome to start testing it.

Thank you for reporting the issue with the DTSI generation script.  We can investigate that more when we start adding the RT1040 support.

> Additionally we found that the DR_SET/TOGGLE/CLEAR register appear not to work on the current die revision, is NXP aware of this? We attached to the EVK using a Segger and found that even manually writing to those registers had no effect on the actual GPIO. In the end we unasserted that feature in the features.h file 

You have some great comments in your post, but I do not think this HAL PR is the best place to discuss some of these.  I created a [GPIO on NXP MIMXRT1040-EVK](https://github.com/zephyrproject-rtos/zephyr/discussions/56777) to continue this discussion there.  Please review that post, and I look forward to your feedback there.

Best regards

**Review by dleach02 — APPROVED** [review]

**Comment by mmahadevan108** [comment]
@yves, could you please update the Zephyr PR with this SHA

**Comment by JiafeiPan** [comment]
Hi, @yvanderv, may I know this PR is to update nxp_hal to mcux-sdk 2.13.0 or 2.13.1?  thanks.

**Comment by DerekSnell** [comment]
> NXP plans to add board support for the MIMXRT1040-EVK to the Zephyr repo in time for the Zephyr v3.4 release. When the PR is created for the initial support, you are very welcome to start testing it.

@matt-wood-ct , FYI we have this Pull Request now adding support for MIMXRT1040-EVK.  Enjoy!
https://github.com/zephyrproject-rtos/zephyr/pull/57358

**Comment by matt-wood-ct** [comment]
> > NXP plans to add board support for the MIMXRT1040-EVK to the Zephyr repo in time for the Zephyr v3.4 release. When the PR is created for the initial support, you are very welcome to start testing it.
> 
> @matt-wood-ct , FYI we have this Pull Request now adding support for MIMXRT1040-EVK. Enjoy! [zephyrproject-rtos/zephyr#57358](https://github.com/zephyrproject-rtos/zephyr/pull/57358)

Hooray, I will indeed starting taking a look great to see things moving many thanks!

**Pull request merged**